pkg/apply/apply_test.go

```diff
@@ -66,41 +88,72 @@ func TestApplyBasicUpdates(t *testing.T) {
 	// Update retention and settings
 	applier.topicConfig.Spec.RetentionMinutes = 400
 	applier.topicConfig.Spec.Settings["cleanup.policy"] = "delete"
-	err = applier.Apply(ctx)
+	applier.topicConfig.Spec.Settings["max.message.bytes"] = "600000"
+	changes, err = applier.Apply(ctx)
 	require.NoError(t, err)
 	topicInfo, err = applier.adminClient.GetTopic(ctx, topicName, true)
 	require.NoError(t, err)
 
+	// test NewOrUpdatedChanges has expected shape when updating configs
+	expectedChanges = &NewOrUpdatedChanges{
+		NewChanges: nil,
+		UpdateChanges: &UpdateChangesTracker{
+			Action: "update",
+			Topic:  topicName,
+			NewConfigEntries: &[]NewConfigEntry{
+				{
+					Name:  "max.message.bytes",
+					Value: "600000",
+				},
+			},
+			UpdatedConfigEntries: &[]ConfigEntryChanges{
+				{
+					Name:    "cleanup.policy",
+					Current: "compact",
+					Updated: "delete",
+				},
+				{
+					Name:    "retention.ms",
+					Current: "30000000",
+					Updated: "27000000",
+				},
+			},
+			MissingKeys: []string{},
+		},
+	}
+	fmt.Printf("EXPECTED: %#v\n\n", expectedChanges.UpdateChanges.UpdatedConfigEntries)
+	fmt.Printf("ACTUAL: %#v\n\n", changes.UpdateChanges.UpdatedConfigEntries)
```

Comment:
Please do not print in the test, use assertions to communicate the results.

Reply:
whoops, this wasn't supposed to end up here, good catch